core/src/main/scala/kafka/server/ConfigHelper.scala

```diff
@@ -57,13 +58,15 @@ class ConfigHelper(metadataCache: MetadataCache, config: KafkaConfig, configRepo
           authHelper.authorize(request.context, DESCRIBE_CONFIGS, CLUSTER, CLUSTER_NAME)
         case ConfigResource.Type.TOPIC =>
           authHelper.authorize(request.context, DESCRIBE_CONFIGS, TOPIC, resource.resourceName)
+        case ConfigResource.Type.CLIENT_METRICS =>
+          authHelper.authorize(request.context, DESCRIBE_CONFIGS, CLIENT_METRICS, resource.resourceName)
```

Comment:
The KIP says that the ACL checked is operation `DESCRIBE_CONFIGS` and resource `CLUSTER`.

Reply:
Completely missed it, thanks for that. I have removed declaring new Resource Type for client_metrics and corresponding checks. Have changed test cases to authorize on CLUSTER resource for CLIENT_METRICS config source.